static/app/components/events/groupingInfo/index.tsx

```diff
@@ -22,116 +23,66 @@ const groupingFeedbackTypes = [
-            evidence: {
-              op: evidenceData?.op,
-              parent_span_ids: evidenceData?.parentSpanIds,
-              cause_span_ids: evidenceData?.causeSpanIds,
-              offender_span_ids: evidenceData?.offenderSpanIds,
-            },
+function generatePerformanceGroupInfo({event, group}: {event: Event; group: Group}) {
+  const {occurrence} = event;
+  const {evidenceData} = occurrence as EventOccurrence;
+
+  const variant = group
+    ? {
+        [group.issueType]: {
+          description: t('performance problem'),
+          hash: occurrence?.fingerprint[0] || '',
+          hasMismatch: false,
+          key: group.issueType,
+          type: 'performance-problem',
+          evidence: {
+            op: evidenceData?.op,
+            parent_span_ids: evidenceData?.parentSpanIds,
+            cause_span_ids: evidenceData?.causeSpanIds,
+            offender_span_ids: evidenceData?.offenderSpanIds,
           },
-        }
-      : null;
-
-    return variant;
-  }
-
-  renderGroupInfoSummary() {
-    const {groupInfo: _groupInfo} = this.state;
-    const {group, event} = this.props;
-
-    const groupInfo =
-      group?.issueCategory === IssueCategory.PERFORMANCE &&
-      event.occurrence &&
-      event.type === 'transaction'
-        ? // performance issue grouping details are generated clint-side
-          this.generatePerformanceGroupInfo()
-        : _groupInfo;
-
-    const groupedBy = groupInfo
-      ? Object.values(groupInfo)
-          .filter(variant => variant.hash !== null && variant.description !== null)
-          .map(variant => variant.description)
-          .sort((a, b) => a!.toLowerCase().localeCompare(b!.toLowerCase()))
-          .join(', ')
-      : t('nothing');
-
-    return (
-      <p data-test-id="loaded-grouping-info">
-        <strong>{t('Grouped by:')}</strong> {groupedBy}
-      </p>
-    );
-  }
+        },
+      }
+    : null;
 
-  renderGroupConfigSelect() {
-    const {configOverride} = this.state;
-    const {event, organization} = this.props;
+  return variant;
+}
 
+export function EventGroupingInfo({
+  event,
+  projectSlug,
+  showGroupingConfig,
+  group,
+}: GroupingInfoProps) {
+  const organization = useOrganization();
+  const [isOpen, setIsOpen] = useState(false);
+  const [configOverride, setConfigOverride] = useState(null);
+
+  const hasPerformanceGrouping =
+    event.occurrence &&
+    group?.issueCategory === IssueCategory.PERFORMANCE &&
+    event.type === 'transaction';
+
+  const {data, isLoading, isError, isSuccess} = useApiQuery<EventGroupInfo>(
+    [
+      `/projects/${organization.slug}/${projectSlug}/events/${event.id}/grouping-info/`,
+      {query: configOverride ? {config: configOverride} : {}},
+    ],
+    {enabled: !hasPerformanceGrouping, staleTime: Infinity}
+  );
+
+  const groupInfo = hasPerformanceGrouping
+    ? generatePerformanceGroupInfo({group, event})
+    : data;
+
+  const renderGroupConfigSelect = () => {
```

Comment:
Can we surround this function with `useCallback`?

Reply:
Extracted the render functions to their own components